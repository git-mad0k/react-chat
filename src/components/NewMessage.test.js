/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewMeesage from './NewMessage';

jest.mock('@material-ui/core/Button', () => () => 'Button');

const mockData = {
  showJoinBtn: true,
  onSendMessage: jest.fn(),
  disabled: false,
  onJoinChat: jest.fn(),
};

describe('NewMeesage', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewMeesage {...mockData} />, div);
  });

  it('render correctly', () => {
    const tree = renderer.create(<NewMeesage {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render showJoin false', () => {
    const tree = renderer.create(<NewMeesage {...mockData} showJoinBtn={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
