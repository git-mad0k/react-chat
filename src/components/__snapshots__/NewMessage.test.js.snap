// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMeesage render correctly 1`] = `
<div
  className="NewMessage-newMessage-1"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3"
  >
    Button
  </div>
</div>
`;

exports[`NewMeesage render showJoin false 1`] = `
<div
  className="NewMessage-newMessage-1"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation2-6 MuiPaper-rounded-3"
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-29 MuiFormControl-marginNormal-30 MuiFormControl-fullWidth-32"
      onKeyPress={[Function]}
      style={
        Object {
          "padding": 8,
        }
      }
    >
      <div
        className="MuiInputBase-root-46 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-fullWidth-55 MuiInput-fullWidth-40 MuiInputBase-formControl-47 MuiInput-formControl-34"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInputBase-input-56 MuiInput-input-41"
          defaultValue={undefined}
          disabled={false}
          id="standard-full-width"
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Type your message..."
          readOnly={undefined}
          required={false}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
